Feature/floorplan

km1tch opened the pull request:
Prob want this INSTEAD of PR4 (since i don't think i based this right)

km1tch committed:
Refactor: Modularization of JavaScript codebase

- Move pricing.js to lib/pricing.js and update all references
- Create new lib modules: api.js, packing.js, location.js
- Add organized CSS structure in assets/css/
- Add component modules for forms and tables
- Update main.py to serve modules from lib directory
- Tweaks to price editor
- Add icon assets for price editor
- Update all import paths to use new module structure

km1tch committed:
Add print support.  Fix minor bug in undo

km1tch committed:
Track location of boxes on a per-store floorplan.  Also print support for price-editor (which will probably morph soon to both a R/W editor and a R/O table view

km1tch committed:
Add simple auth (in place of editable flag in YAML).  Minor tweaks to price_editor.  Cleaned up folder locations a smidge

navigator9951 commented:
Not sure if you got my message, on Reddit, but I'll be back home from vacation in a few days to merge everything. I promise I'm not gone or anything.

km1tch committed:
Fix dimension sorting bug in Box constructor

Same fix as applied to upstream/main - ensures correct tracking of open dimension after sorting
